evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -1072,7 +1072,7 @@ def test_binary_predict_pipeline_objective_mismatch(
         ValueError,
         match="Objective Precision Micro is not defined for time series binary classification.",
     ):
-        binary_pipeline.predict(X[30:32], "precision micro", X[:30], y[:30])
```

Comment:
What's this change for? Is it because we can't infer frequencies with only two values? Should we handle this somewhere?

Reply:
That's correct - pandas needs 3 or more values to infer a frequency. I'm not sure where we could handle this, though.

Reply:
OK fair enough. How does this affect our ability to forecast out 1 unit? is it still possible?

Reply:
@jeremyliweishih we infer from y_train in the decomposer's fit, so the pipeline has no frequency knowledge.

@fjlanasa that was a good catch - this change would have prevented us from forecasting out 1 unit. Fortunately, the frequency detection isn't actually required within drop_time_index thanks to the other changes here, so I've just reverted it.